pandas/io/pytables.py

```diff
@@ -3020,12 +3050,50 @@ def set_info(self):
         """ update our table index info """
         self.attrs.info = self.info
 
+    def set_non_index_axes(self):
+        """ Write the axes to carrays """
+        def f(dim, flds):
+            name = "non_index_axes_%d" % dim
+            try:
+                self._handle.create_carray(self.attrs._v_node, name, obj=np.array(flds))
```

Comment:
use ``_convert_index`` on the data (and ``_unconvert_index`` on deserialize); this will do all the proper type conversions